slidechain/watch.go

```diff
@@ -117,56 +117,97 @@ func (c *Custodian) watchExports(ctx context.Context) {
 		}
 		b := got.(*bc.Block)
 		for _, tx := range b.Transactions {
-			// Look for a retire-type ("X") entry
-			// followed by a specially formatted log ("L") entry
+			// Check if the transaction has either expected length for an export tx.
+			// Confirm that its input, log, and output entries are as expected.
+			// If so, look for a specially formatted log ("L") entry
 			// that specifies the Stellar asset code to peg out and the Stellar recipient account ID.
+			if len(tx.Log) != 4 && len(tx.Log) != 6 {
+				continue
+			}
+			if tx.Log[0][0].(txvm.Bytes)[0] != txvm.InputCode {
+				continue
+			}
+			if tx.Log[1][0].(txvm.Bytes)[0] != txvm.LogCode {
+				continue
+			}
 
-			for i := 0; i < len(tx.Log)-2; i++ {
-				item := tx.Log[i]
-				if item[0].(txvm.Bytes)[0] != txvm.RetireCode {
-					continue
-				}
-				retiredAmount := int64(item[2].(txvm.Int))
-				retiredAssetIDBytes := []byte(item[3].(txvm.Bytes))
+			outputIndex := len(tx.Log) - 2
+			if tx.Log[outputIndex][0].(txvm.Bytes)[0] != txvm.OutputCode {
```

Comment:
To be sure this is an export tx (and not a tx masquerading as one), it would be good to check that the export-contract seed appears in the log. The easiest way to do that is to have the export contract do a `log` instruction and then checking the seed in the extra `{"L", seed, item}` entry against `exportContract1Seed`.

This can be for a future PR, but should be at least a TODO in this one.